pkgs/shells/fish/default.nix

```diff
@@ -270,14 +263,6 @@ let
           -i $out/share/fish/functions/{__fish_print_packages.fish,__fish_print_addresses.fish,__fish_describe_command.fish,__fish_complete_man.fish,__fish_complete_convert_options.fish} \
              $out/share/fish/completions/{cwebp,adb,ezjail-admin,grunt,helm,heroku,lsusb,make,p4,psql,rmmod,vim-addons}.fish
 
-    '' + optionalString usePython ''
```

Comment:
Python is also used by many auto-completions (including [composer](https://github.com/fish-shell/fish-shell/pull/6225)). But all those and `fish_config` find Python using the `__fish_anypython` function, which looks like it should be able to [discover Python from `PATH`](https://github.com/fish-shell/fish-shell/blob/9af6a64fd2bd132ac1866af846fdc8368e086206/share/functions/__fish_anypython.fish#L5-L6).

`share/fish/tools/create_manpage_completions.py` currently uses `#!/usr/bin/env python3` in shebang so that is not affected either.

So looks like this PR will only make it take slightly longer to find Python and break those features in environments without Python, not sure if any exist in practice.

Reply:
I opened this PR to received feedback such as yours. Now I will adjust the PR to make something usable and lighter.